### PR TITLE
Speed up CSV export by instantiating dataset with data and headers

### LINKED ISSUE #2650: Increase speed of CSV export
### Describe the Bug

#2593 introduced the option to export region feedback to CSV. However the testing on production and test server showed that in it's current state the export is very slow (~17s on production).

### Steps to Reproduce

1. Go to Testumgebung on the production server
2. Click on Feedback
3. Select an arbitrary amount of feedback entries
4. Open up network analysis in browser
5. Run the bulk action "export to CSV"
6. See that the action takes a few seconds

### Expected Behavior

I'm not sure what a good threshold would be - maybe the standard 4s?

### Actual Behavior

It takes considerably longer - around 17s in my test cases.

### Additional Information

While testing this I noticed that the amount of selected feedback didn't really have an effect on the performance. Exporting 1 and 16 pages took the same loading time (the above mentioned 17s). I'm not sure if this is relevant, but I still wanted to mention it.

<details>
  <summary>Traceback</summary>

  ```
  ```

</details>

### PR DESCRIPTION
**Pull request by cclauss**
### Short description

Fixes #2650
* #2650

https://tablib.readthedocs.io/en/stable/api.html suggests that:
> You can also set rows and headers upon instantiation.

### Proposed changes

- Instantiate the `Dataset` with all data up front instead of repeatedly calling `Dataset.append()`
- Also, use [`Dataset.export("csv")`](https://tablib.readthedocs.io/en/stable/api.html#tablib.Dataset.export) which is documented instead of `Dataset.csv` which is not.

### Side effects

- I do not know how to test or benchmark this beyond the current pytests.
- 

### Resolved issues

Fixes: #2650

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Comment by cclauss** [comment]
@timobrembeck Can you please unblock these so we can see if the tests pass with the proposed changes?
> setup-test-reporter - Unauthorized
> test - Blocked

File: integreat_cms/cms/views/feedback/region_feedback_actions.py
Comment (PeterNerlich) [inline integreat_cms/cms/views/feedback/region_feedback_actions.py:203]: I think these should be unpacked as args
```suggestion
        *(resource.export_resource(obj) for obj in selected_feedback),
```

**Review by PeterNerlich — APPROVED** [review]
So I have no clue about the process for making the CI work, I guess @timobrembeck  has probably the power to override these checks or whatever.

I tested the changes and ran the test.sh locally, and except for a handful of PDF tests that always make issues and are very unlikely to be affected by this change, everything seems perfect, so I'm just approving this PR right now.

**Review by ulliholtgrave — APPROVED** [review]
I restarted the checks with my permission and everything seems fine now. Thanks for your contribution! 😊

**Comment by cclauss** [comment]
Can you please try with #2679 as well?

**Comment by PeterNerlich** [comment]
@timobrembeck What is your opinion, can we merge this? I think your last question on the team call was whether this actually fixes the performance issues – I think the performance issues are likely rooted deeper in the system, but this PR is an improvement anyway and I don't see any reason to delay it. We can remove the *Fixes:* line or just reopen the issue if your worry was only about it getting forgotten.

**Comment by PeterNerlich** [comment]
@cclauss While waiting on an answer, could you please manually squash your 3 commits into one?

**Comment by PeterNerlich** [comment]
No, that is disabled in this repository (don't ask me for what the rationale behind this was originally and why exactly single commits followed by a merge commit are usually preferred here).

I meant to squash it manually e.g. using `git rebase -i` and force pushing onto this branch.

**Comment by cclauss** [comment]
`git rebase -i` is difficult to achieve on my iPhone...  Now I am back to my Mac.

**Pull request merged**
